src/portable/nordic/nrf5x/dcd_nrf5x.c

```diff
@@ -435,8 +530,31 @@ void dcd_int_handler(uint8_t rhport)
     dcd_event_bus_signal(0, DCD_EVENT_BUS_RESET, true);
   }
 
+  // ISOIN: Data was moved to endpoint buffer, client will be notified in SOF
+  if ( int_status & USBD_INTEN_ENDISOIN_Msk )
+  {
+    xfer_td_t* xfer = get_td(8, TUSB_DIR_IN);
+
+    xfer->actual_len = NRF_USBD->ISOIN.AMOUNT;
+    // Data in endpoint has been consumed, client will be notified in SOF that may be
+    // at later time.
+    xfer->data_received = true;
```

Comment:
ahh, I think you want to reuse the variable, but `data_received` is in the view of the device on OUT transfer. This usage is seem to be on opposite direction which should be named `data_consumed`. 

Maybe we should rename it to something neutral, or we can make it a union. That is OK if you don't come up the name ( I have difficulty with naming as well), I just want to confirm the usage of it.

Reply:
correct, I will change it to ```data_consumed``` that should be good for both cases

Reply:
@kasjer ah, not actually, currently the `data_received` is indicator for OUT endpoint, since nrf controller will auto ACK the first OUT packet  even if we didn't schedule any transfer in the software. Therefore it is more like `data_available` for CBI than `consumed`, there for the consumed is not really correct for CBI usage. These naming is really difficult :weary: